discord/ui/select.py

```diff
@@ -106,6 +106,10 @@ def __init__(
     ) -> None:
         super().__init__()
         self._selected_values: List[str] = []
+
+        if not (isinstance(custom_id, str) or custom_id is None):
+            raise TypeError(f"expected custom_id to be str, not {type(custom_id).__name__}")
```

Comment:
Same with this one,
`type(custom_id).__name__}`
should be
`custom_id.__class__.__name__`

Reply:
No, this would cause the following result:

`TypeError: expected custom_id to be str, not type`

Reply:
Nevermind, I misread. 

Regardless, either option has the same result. What's the benefit of your approach?

Reply:
Haha, no worries. I just think it's better form, seems to be more conventional to do it that way (even though they're essentially the same), I'd also say that type has to do a global lookup, while class is much better efficiency-wise (believe it's O(1) from a hashmap?), either way, it should be more efficient/faster and is more conventional

Reply:
Good points, I've applied the changes. Thanks!

Reply:
No problem